cellprofiler/library/functions/image_processing.py

```diff
@@ -197,7 +217,7 @@ def get_adaptive_threshold(
             j0 = int(j * increment[1])
             j1 = int((j + 1) * increment[1])
             block = image[i0:i1, j0:j1]
-            block = block[~numpy.isnan(block)]
+            block = block[~numpy.logical_not(block)]
```

Comment:
I've altered the adaptive thresholding of CellProfiler slightly.

The reason for changing `numpy.isnan` to `numpy.logical_not` is to make how CellProfiler deals with masks more similar to that in skimage, where masked pixels are False.

Additionally, this change is because `get_adaptive_threshold` makes a call to `get_global_threshold`. The former accepted numpy.nan pixels since they were filtered out in the line highlighted here.

However, since `get_adaptive_threshold` makes a call to `get_global_threshold`, the latter is fed NaN pixels. NaN pixels are not a valid input to skimage functions. There's a discussion about scikit-image and NaN values as input here: https://github.com/scikit-image/scikit-image/issues/1624